File: rdds/accumulator.scala
object reload extends App{

    val sc = new SparkContext(local[*], "accumalator")
    val input = sc.textFile("c:\\Users\\pp255070\\OneDrive - Teradata\\Documents\\sample.txt")

    val myaccum = sc.longAccumulator("blank lines accumalator")  // long is data type of accumalator, it is at cluster level that is why sc

    input.foreach(x => if (x=="") myaccum.add(1))

    myaccum.value
}

File: rdds/accumulator.py
from pyspark import SparkContext

def lenaccum(line):
    if len(line) == 0:
        myaccum.add(1)

sc = SparkContext("local[*]", "accum")

myrdd = sc.textFile("path")

myaccum = sc.accumulator(0)  # sc.accumulator(0.0) there are two types of accumulator int, float

myrdd.foreach(lenaccum)

print(myaccum.value)